gems/activerecord/6.0/activerecord.rbs

```diff
@@ -724,10 +724,700 @@ end
 
 module ActiveRecord
   module Associations
-    class CollectionProxy
-      def build: (?::Hash[untyped, untyped] attributes) -> untyped | ...
-      def create: (?::Hash[untyped, untyped] attributes) -> untyped | ...
-      def create!: (?::Hash[untyped, untyped] attributes) -> untyped | ...
+    # Collection proxies in Active Record are middlemen between an
+    # <tt>association</tt>, and its <tt>target</tt> result set.
+    #
+    # For example, given
+    #
+    #   class Blog < ActiveRecord::Base
+    #     has_many :posts
+    #   end
+    #
+    #   blog = Blog.first
+    #
+    # The collection proxy returned by <tt>blog.posts</tt> is built from a
+    # <tt>:has_many</tt> <tt>association</tt>, and delegates to a collection
+    # of posts as the <tt>target</tt>.
+    #
+    # This class delegates unknown methods to the <tt>association</tt>'s
+    # relation class via a delegate cache.
+    #
+    # The <tt>target</tt> result set is not loaded until needed. For example,
+    #
+    #   blog.posts.count
+    #
+    # is computed directly through SQL and does not trigger by itself the
+    # instantiation of the actual post records.
+    class CollectionProxy < Relation
+      interface _EachPair
+        def each_pair: () { ([untyped, untyped]) -> untyped } -> self
```

Comment:
Is `#empty?` not necessary?

https://github.com/rails/rails/blob/2ebb508cd8ee16c0bb280f91b93c01d939fcbf12/activemodel/lib/active_model/attribute_assignment.rb#L32

And `#each` ?

https://github.com/rails/rails/blob/2ebb508cd8ee16c0bb280f91b93c01d939fcbf12/activemodel/lib/active_model/attribute_assignment.rb#L41

Reply:
I'm not sure what is the best way for such an interface. But no reason to ignore them. Okay, I'll add them soon.

Reply:
Done. I merged https://github.com/ruby/gem_rbs_collection/pull/603 into this PR to support `ActionController::Parameters`.